src/main/DinerMenuIterator.java

```diff
@@ -0,0 +1,28 @@
+package main;
+
+import java.util.Iterator;
+
+public class DinerMenuIterator implements Iterator {
+    MenuItem[] list;
+    int position = 0;
+
+    public DinerMenuIterator(MenuItem[] items) {
+        this.list = items;
+    }
+
+    @Override
+    public boolean hasNext() {
+        if (position >= list.length || list[position] == null) {
```

Comment:
I would definitely prefer to see this negated so you can get rid of the `if` and simply return the conditional expression instead.